src/components/trade/PriceSuggestions/PriceSuggestionWidget.tsx

```diff
@@ -41,17 +41,20 @@ export const PriceSuggestionWidget: React.FC<Props> = (props) => {
   const { setValue, trigger } = useFormContext<TradeFormData>()
 
   const updatePrices = useCallback(
-    (price: string, invertedPrice) => {
+    async (price: string, invertedPrice) => {
       if (isPriceInverted) {
         setValue(priceInverseInputId, price)
         setValue(priceInputId, invertedPrice)
       } else {
         setValue(priceInputId, price)
         setValue(priceInverseInputId, invertedPrice)
       }
-      trigger()
+      // stupid hack to fix validation
+      // otherwise trigger doesnt validate properly the new values
+      await new Promise((accept) => setTimeout(accept, 0))
```

Comment:
I don't get it

cc/ @Velenir

Reply:
Does it trigger sync with old values? @W3stside 
What about `useEffect(trigger, [isPriceInverted, priceInverseInputId, priceInputId])`

Reply:
not sure @Velenir didn't test comprehensively honestly, just added this knowing ti would work :sweat: i can try again later